src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -612,13 +548,13 @@ private void removeTerritoriesThatArentWorthAttacking(final List<ProTerritory> p
         int minCost = Integer.MAX_VALUE;
         Unit minUnit = null;
         for (final Unit u : t.getUnits().getMatches(Matches.unitIsOwnedBy(player))) {
-          if (playerCostMap.getInt(u.getType()) < minCost) {
-            minCost = playerCostMap.getInt(u.getType());
+          if (ProData.unitValueMap.getInt(u.getType()) < minCost) {
```

Comment:
So the data is essentially immutable (at least during a single phase). At the start of each phase, the data classes is initialized and then it holds the current game/map/player state for all the client classes to use so that they don't need to all pass around data/map/player/etc.

At this point, I was keeping ProData and ProUtilities separate though I could just combine them.